packages/minimongo/minimongo.js

```diff
@@ -782,8 +782,22 @@ LocalCollection.prototype.update = function (selector, mod, options, callback) {
   // generate an id for it.
   var insertedId;
   if (updateCount === 0 && options.upsert) {
-    var newDoc = LocalCollection._removeDollarOperators(selector);
+    var newDoc = {};
+     
+    if (LocalCollection._selectorIsId(selector)) {
+      selector = {_id: selector}
+    }
+
+    var selectorDocument = LocalCollection._removeDollarOperators(selector);
```

Comment:
Good points; naming things is hard! Okay, I'm good with your suggestion of `selectorModifier` or `upsertModifier`. I agree that `selectorModifier` seems a bit strange, but since that's really what it is, that might be best.